internal/config/cassandra.go

```diff
@@ -209,13 +216,15 @@ func (cfg *Cassandra) Opts() (opts []cassandra.Option, err error) {
 
 	if cfg.Net != nil {
 		der, err := net.NewDialer(cfg.Net.Opts()...)
-		if err == nil {
-			opts = append(opts,
-				cassandra.WithDialer(
-					der,
-				),
-			)
+		if err != nil {
+			return nil, err
```

Comment:
I'm not sure but is it okay to return `nil` here?
At L237, even if an error occurred at the line before it, it returns `opts`. What's the difference between them?

Reply:
I think you are correct, but I think we need to return the options even when the error occurred.
e.g.
```go
	if cfg.Net != nil {
		der, err := net.NewDialer(cfg.Net.Opts()...)
                if err == nil {
                      opts = append(opts,
			cassandra.WithDialer(
				der,
			),
		)
            } else {
                   errs = errors.Wrap(errs, err.Err())
           }
       }

	if cfg.TLS != nil && cfg.TLS.Enabled {
		tcfg, err := tls.New(
			tls.WithCert(cfg.TLS.Cert),
			tls.WithKey(cfg.TLS.Key),
			tls.WithCa(cfg.TLS.CA),
		)
		if err != nil {
			return opts, errors.Wrap(errs, err)
		}
```

Reply:
@rinx thanks for the review.
Actually, I'm not sure whether which is best.
I just fix this along with `internal/config/mysql.go`.
https://github.com/vdaas/vald/blob/master/internal/config/mysql.go#L110-L113

But, if `net.newDialer()` returns the error, we cannot connect to the Cassandra cluster (is it right?).
If it is so, I think returning an error is not bad to prevent other errors, and throwing away the opts that are created before is too.

What do you think about this?
I think both are not bad and I'd like to get your suggestion as an author. 
cc: @kpango

Reply:
To tell the truth, we don't have any obvious policy to handle problems like this, so i cannot answer it immediately.

> But, if net.newDialer() returns the error, we cannot connect to the Cassandra cluster (is it right?).

Even if it cannot get a `der` from `newDialer()`, internal/cassandra will work with the default dialer of gocql.

Reply:
But even if `internal/cassandra` works with the default dialer, it may not work as excepted from the users...

Reply:
Sure.
There's the same problem about L237.

So we should think about how to handle problems like this.

Reply:
The previous implementations had a policy of implicitly using the Default Dialer without returning an error, but since it seems that implicit continuous should not be done, I thought it would be best practice for the API definition to return an error and ask the user for an appropriate error handle, which is why I LGTM'd this fix.

However, there has been no discussion on whether or not to return the return value of origin when returning an error.
Normally, it is better to return zero or empty value in case of an error, so I personally think it is better to follow this rule.

Reply:
ok, it seems we have to discuss this and get the way how we handle it.
Depending on the discussion, we have to update `internal/config/mysql.go#L237` too.

Reply:
should we also return `nil, error` on line 238?
https://github.com/vdaas/vald/pull/1117/files#diff-82743e089bd2afaf22dfee471faeb927c8f7c87d9f392badc6c5d82431e83680L228

Reply:
`However, there has been no discussion on whether or not to return the return value of origin when returning an error.`
I think we should discuss it.

Reply:
I think there are 3 options for this problem.
1. ignore the error
    - it does not seems good handling 😅 
2. return error but other functional options are still alive, and output Error (or Warn) Log.
    - if the error does not cause a critical issue on the Vald, it can be.
    - user will get the configurations error via each component's log and then can fix it.
3. return error and all of the options are restored.
    - if the error will have a fatal effect on running the Vald, we should stop the applying configuration.

Maybe we have to use properly option 2 and 3 based on the situation?
If it is yes, we need to define the policy.
What do you think?

Reply:
3